### PR TITLE
Introduce CI FormatCheck

### PR DESCRIPTION
**Pull request by ytdHuang**
In this PR, I replace the original Auto-JuliaFormatter-PR CI as Format-Check.

This is just a simple check whether the code is formatted before merging.

### CODE REVIEW

**Comment by albertomercurio** [comment]
Why don’t we use the proposed code of the action webpage?

```yaml
name: Format suggestions
on:
  pull_request:
    # this argument is not required if you don't use the `suggestion-label` input
    types: [ opened, reopened, synchronize, labeled, unlabeled ]
jobs:
  code-style:
    runs-on: ubuntu-latest
    steps:
      - uses: julia-actions/julia-format@v3
        with:
          version: '1' # Set `version` to '1.0.54' if you need to use JuliaFormatter.jl v1.0.54 (default: '1')
          suggestion-label: 'format-suggest' # leave
```

BTW, we should try to avoid to rename CIs, otherwise we accumulate all the names in the Actions page

**Comment by ytdHuang** [comment]
To avoid conflicts, I will push the formatted codes after PR #117 is merged.

**Comment by ytdHuang** [comment]
> Why don’t we use the proposed code of the action webpage?
> 
> ```yaml
> name: Format suggestions
> on:
>   pull_request:
>     # this argument is not required if you don't use the `suggestion-label` input
>     types: [ opened, reopened, synchronize, labeled, unlabeled ]
> jobs:
>   code-style:
>     runs-on: ubuntu-latest
>     steps:
>       - uses: julia-actions/julia-format@v3
>         with:
>           version: '1' # Set `version` to '1.0.54' if you need to use JuliaFormatter.jl v1.0.54 (default: '1')
>           suggestion-label: 'format-suggest' # leave
> ```
> 
> BTW, we should try to avoid to rename CIs, otherwise we accumulate all the names in the Actions page

They mentioned some issues in their [README](https://github.com/julia-actions/julia-format?tab=readme-ov-file). I think it would be better for us to just write it by ourselves.

**Comment by ytdHuang** [comment]
I have disabled the outdated CI workflows in action page

File: .github/workflows/FormatCheck.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - synchronize
+      - ready_for_review
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@v2
```
Comment (albertomercurio) [inline .github/workflows/FormatCheck.yml:23]: Can we use the cache action, or not?

**Review by albertomercurio — APPROVED** [review]

File: .github/workflows/FormatCheck.yml (hunk 1 above)
Comment (albertomercurio) [reply .github/workflows/FormatCheck.yml:23]: Otherwise we can merge.

Comment (ytdHuang) [reply .github/workflows/FormatCheck.yml:23]: not sure whether need the cache, since this CI only need to install `JuliaFormatter.jl`

**Comment by ytdHuang** [comment]
The JuliaFormatter keep formatting `README.md` into a wrong one.

**Comment by albertomercurio** [comment]
The format check is failing. Is that ok?

**Comment by albertomercurio** [comment]
Mmmm that’s strange

**Comment by ytdHuang** [comment]
> The format check is failing. Is that ok?

The only file that is not formatted now is `README`, but `JuliaFormatter` will format it incorrectly.

**Comment by ytdHuang** [comment]
> Ok, so let’s merge?

I add an `ignore = ["README.md"]` to `.JuliaFormatter.toml`.

**Comment by albertomercurio** [comment]
Ok we can merge it

**Pull request merged**